### PR TITLE
NO-ISSUE: e2e tests refactoring and kind adoption

### PR DESCRIPTION
**Pull request by andfasano**
This patch reviews the current approach used for running e2e tests, by reusing the native kind support already offered by e2e-framework. 
In addition, manifests generation have been revisited for an improved alignment with the ones used in production.

For running the tests locally:

```
$ make e2e-tests
```

### CODE REVIEW

**Comment by derekhiggins** [comment]
Do I need to install kind?

```
$ make e2e-tests 
go test ./tests/e2e/...
2024/05/27 18:17:16 Creating cluster (ofcir-test)
F0527 18:17:17.001808 2187241 env.go:369] Setup failure: kind not available even after installation
FAIL	github.com/openshift/ofcir/tests/e2e	0.208s
FAIL
make: *** [Makefile:82: e2e-tests] Error 1
```

**Comment by andfasano** [comment]
> Do I need to install kind?
> 
> ```
> $ make e2e-tests 
> go test ./tests/e2e/...
> 2024/05/27 18:17:16 Creating cluster (ofcir-test)
> F0527 18:17:17.001808 2187241 env.go:369] Setup failure: kind not available even after installation
> FAIL	github.com/openshift/ofcir/tests/e2e	0.208s
> FAIL
> make: *** [Makefile:82: e2e-tests] Error 1
> ```

That sghouldn't be necessary, as the setup function will install it if not found

**Comment by andfasano** [comment]
> When the job did fail (because kubectl was missing), the kind container was left behind running, could this be cleaned up on failure?

There is a known bug in the current e2e-framework (the teardown is not correctly invoked in case of failure), but it has been fixed only in the latest version (0.3.0). 
I didn't want also to upgrade all the libs/dependencies in this PR (due the additional amount of changes required), but we could followup after we landed it

**Comment by derekhiggins** [comment]
>>ln -s /home/derekh/bin/oc /home/derekh/bin/kubectl

> Addressed in the CI PR https://github.com/openshift/release/pull/52425

Ah ok, would it be possible to test for kubectl in this repo so that users running the tests locally know why its failing ?

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
 generatorOptions:
   disableNameSuffixHash: true
 
-configMapGenerator:
-- files:
-  - controller_manager_config.yaml
-  name: manager-config
-apiVersion: kustomize.config.k8s.io/v1beta1
-kind: Kustomization
+# # Generate a ConfigMap from the specified file
```
[the remark this answers is not in the document]
Comment (andfasano) [reply config/manager/kustomization.yaml:12]: Not really, just thought to keep them for now and eventually perform a deeper cleanup later (by removing also the not used fkustomization files).

**Comment by derekhiggins** [comment]
> > > ln -s /home/derekh/bin/oc /home/derekh/bin/kubectl
> 
> > Addressed in the CI PR [openshift/release#52425](https://github.com/openshift/release/pull/52425)
> 
> Ah ok, would it be possible to test for kubectl in this repo so that users running the tests locally know why its failing ?

@andfasano any thoughts on this?

**Comment by andfasano** [comment]
> > > > ln -s /home/derekh/bin/oc /home/derekh/bin/kubectl
> > 
> > 
> > > Addressed in the CI PR [openshift/release#52425](https://github.com/openshift/release/pull/52425)
> > 
> > 
> > Ah ok, would it be possible to test for kubectl in this repo so that users running the tests locally know why its failing ?
> 
> @andfasano any thoughts on this?

Shouldn't be already a failure message, or am I missing something?

**Comment by derekhiggins** [comment]
> > > > > ln -s /home/derekh/bin/oc /home/derekh/bin/kubectl
> > > 
> > > 
> > > > Addressed in the CI PR [openshift/release#52425](https://github.com/openshift/release/pull/52425)
> > > 
> > > 
> > > Ah ok, would it be possible to test for kubectl in this repo so that users running the tests locally know why its failing ?
> > 
> > 
> > @andfasano any thoughts on this?
> 
> Shouldn't be already a failure message, or am I missing something?

Ah yes there is, 
bash: line 1: kubectl: command not found

So looks like  I was mixing up my errors, the one I think we need some reporting for is if "kind" isn't in PATH, in that case you get  (Fixed by export PATH=$PATH:/home/derekh/go/bin/)

```
go test ./tests/e2e/...
2024/06/12 12:27:36 Creating cluster (ofcir-test)
F0612 12:27:36.424053 4047902 env.go:369] Setup failure: kind not available even after installation
FAIL	github.com/openshift/ofcir/tests/e2e	0.411s
FAIL
make: *** [Makefile:82: e2e-tests] Error 1

```

**Pull request merged**